### PR TITLE
Implement audit changes

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This pull requests implements the options part of the API audit notes.

This is currently very much a work in progress

Related #5225

### CODE REVIEW

File: packages/core/core/src/AssetGraphBuilder.js [hunk 1]
```diff
@@ -818,7 +818,7 @@ export default class AssetGraphBuilder extends EventEmitter {
   }
 
   async readFromCache(): Promise<?Array<Event>> {
-    if (this.options.disableCache) {
+    if (this.options.shouldDisableCache) {
```
Comment (mischnic) [inline packages/core/core/src/AssetGraphBuilder.js:821]: I thought we had talked about this in the call but maybe this should rather be `shouldEnableCache`.

Comment (DeMoorJasper) [reply packages/core/core/src/AssetGraphBuilder.js:821]: All changes have been taken from the linked PR by Devon, which are the changes we agreed on I think.

Comment (mischnic) [reply packages/core/core/src/AssetGraphBuilder.js:821]: Yeah, but at the same time I remember Will saying that double negations should be avoided 🤷‍♂️

**Comment by DeMoorJasper** [comment]
Closing in favor of #5572 will do these changes in smaller incremental PRs this PR will take ages to complete and be a nightmare to keep in sync with v2

File: packages/core/core/src/AssetGraphBuilder.js (hunk 1 above)
Comment (DeMoorJasper) [reply packages/core/core/src/AssetGraphBuilder.js:821]: Yeah it was commented out so probably not final yet

**Pull request closed without merging**